### ISSUE TITLE
Support deploying own ERP container on K8s via ECK

### ISSUE THREAD

**Issue by a03nikki** [elastic/package-registry#783]
The documentation describes how to run a EPR (Elastic Package Registry) using Docker.  It would be great if ECK (Elastic Cloud on Kubernetes) could deploy the container to a Kubernetes (K8s) cluster.

Related documentation: https://www.elastic.co/guide/en/integrations-developer/current/air-gapped.html#air-gapped-diy-epr
Related issue: https://github.com/elastic/cloud-on-k8s/issues/5107

**Comment by Kushmaro** [comment]
Hey @a03nikki :) heard about this and thought I'd investigate 
Is there any specific value in ECK doing this, vs. just deployment of the EPR container regularly on k8s?

**Comment by a03nikki** [comment]
@Kushmaro : ECK could do the configuration for Kibana to use the custom EPR endpoint as the pod would then be controlled by the operator.

**Comment by omlet05** [comment]
On my side I did something like this to run EPR on Kubernetes. Could you add something similar (or better) to the documentation? Or maybe provide a helm deployment?
`apiVersion: apps/v1
kind: Deployment
metadata:
  name: elastic-package-registry-deployment
spec:
  selector:
    matchLabels:
      app: elastic-package-registry
  replicas: 1
  template:
    metadata:
      labels:
        app: elastic-package-registry
    spec:
      containers:
      - name: elastic-package-registry
        image: docker.elastic.co/package-registry/distribution:8.11.1
        imagePullPolicy: Always
        resources:
          limits:
            memory: "1024Mi"
            cpu: 1
          requests:
            memory: "1024Mi"
            cpu: 100m
      securityContext:
        runAsUser: 1000
        runAsGroup: 1000
        fsGroup: 1000
---
apiVersion: v1
kind: Service
metadata:
  name: elastic-package-registry-http
  labels:
    run: elastic-package-registry
spec:
  ports:
  - protocol: TCP
    port: 80
    targetPort: 8080
  selector:
    app: elastic-package-registry`

**Comment by omlet05** [comment]
_On my side I did something like this to run EPR on Kubernetes. Could you add something similar (or better) to the documentation? Or maybe provide a helm deployment?_

`apiVersion: apps/v1
kind: Deployment
metadata:
  name: elastic-package-registry-deployment
spec:
  selector:
    matchLabels:
      app: elastic-package-registry
  replicas: 1
  template:
    metadata:
      labels:
        app: elastic-package-registry
    spec:
      containers:
      - name: elastic-package-registry
        image: docker.elastic.co/package-registry/distribution:8.11.1
        imagePullPolicy: Always
        resources:
          limits:
            memory: "1024Mi"
            cpu: 1
          requests:
            memory: "1024Mi"
            cpu: 100m
      securityContext:
        runAsUser: 1000
        runAsGroup: 1000
        fsGroup: 1000
---
apiVersion: v1
kind: Service
metadata:
  name: elastic-package-registry-http
  labels:
    run: elastic-package-registry
spec:
  ports:
  - protocol: TCP
    port: 80
    targetPort: 8080
  selector:
    app: elastic-package-registry`

**Comment by QuinnBast** [comment]
Agreed. Or even better, [add it to the ECK operator](https://github.com/elastic/cloud-on-k8s/issues/5107)
